### PR TITLE
feat: add translation menu

### PR DESCRIPTION
**Pull request by InnaAtanasova**
This branch is a copy of the `fix/multi-language` branch with cherry-picked commits

### CODE REVIEW

File: fix__fixed_language_switches_and_deploy.patch [hunk 1]
```diff
@@ -0,0 +1,760 @@
+Subject: [PATCH] fix: fixed language switches and deploy
```
Comment (droshev) [inline fix__fixed_language_switches_and_deploy.patch:1]: @kalo2711 can you remove this?

File: netlify.toml [hunk 2]
```diff
@@ -8,9 +8,32 @@
...
+[[redirects]]
+  from = "/products"
+  to = "/de/products"
+  status = 301
+  conditions = {Language = ["de"]}
+
+[[redirects]]
+  from = "/products"
+  to = "/products"
+  status = 200
```
Comment (g-cheishvili) [inline netlify.toml:39]:
```suggestion
[[redirects]]
  from = "/"
  to = "/en"
```

Comment (g-cheishvili) [inline netlify.toml:39]:
```suggestion
[[redirects]]
  from = "/"
  to = "/en"
  status = 301
  conditions = {Language = ["en"]}

[[redirects]]
  from = "/"
  to = "/fr"
  status = 301
  conditions = {Language = ["fr"]}

[[redirects]]
  from = "/"
  to = "/de"
  status = 301
  conditions = {Language = ["de"]}

[[redirects]]
  from = "/"
  to = "/en"
  status = 200
```

I suppose this is with language checks

File: fix__fixed_language_switches_and_deploy.patch (hunk 1 above)
Comment (droshev) [inline fix__fixed_language_switches_and_deploy.patch:1]: THIS FILE should not be checked in

File: src/app/fl-language/fl-language.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (droshev) [inline src/app/fl-language/fl-language.component.spec.ts:1]: what is the purpose of it?

File: src/main.ts [hunk 4]
```diff
@@ -1,3 +1,5 @@
+/// <reference types="@angular/localize" />
```
Comment (droshev) [inline src/main.ts:1]: ?

**Pull request closed without merging**